data_structures/src/chain.rs

```diff
@@ -3422,13 +3480,107 @@ pub fn block_example() -> Block {
     }
 }
 
+pub fn dr_root_superblock_loop_test(sb: SuperBlock, expected_indices: Vec<usize>, expected_lemma_lengths: Vec<usize>, blocks: Vec<Block>, dr_txs: Vec<DRTransaction>){
+    for index in 0..expected_indices.len() {
+        let result = sb
+            .dr_proof_of_inclusion(&blocks, &dr_txs[index])
+            .unwrap();
+        assert_eq!(result.index, expected_indices[index]);
+        assert_eq!(result.lemma.len(), expected_lemma_lengths[index]);
+        let lemma = result
+            .lemma
+            .iter()
+            .map(|h| match *h {
+                Hash::SHA256(x) => Sha256(x),
+            })
+            .collect();
+        let proof = InclusionProof::sha256(result.index, lemma);
+        assert!(proof.verify(
+            dr_txs[index].body.data_poi_hash().into(),
+            sb.data_request_root.into()
+        ));
+    }
+}
+
+pub fn tally_root_superblock_loop_test(sb: SuperBlock, expected_indices: Vec<usize>, expected_lemma_lengths: Vec<usize>, blocks: Vec<Block>, tally_txs: Vec<TallyTransaction>){
+    for index in 0..expected_indices.len() {
+        let result = sb
+            .tally_proof_of_inclusion(&blocks, &tally_txs[index])
+            .unwrap();
+        assert_eq!(result.index, expected_indices[index]);
+        assert_eq!(result.lemma.len(), expected_lemma_lengths[index]);
+        let lemma = result
+            .lemma
+            .iter()
+            .map(|h| match *h {
+                Hash::SHA256(x) => Sha256(x),
+            })
+            .collect();
+        let proof = InclusionProof::sha256(result.index, lemma);
+        assert!(proof.verify(
+            tally_txs[index].data_poi_hash().into(),
+            sb.tally_root.into()
+        ));
+    }
+}
+
+pub fn build_test_dr_txs(length: u32) -> Vec<DRTransaction>{
+    let inputs: Vec<Input> = (1..length+1)
```

Comment:
There's a syntax for that:

```suggestion
    let inputs: Vec<Input> = (1..=length)
```

Also, could you move this functions inside `mod tests`? Since they should only be used in tests.